docs/cow-protocol/tutorials/cow-swap/flash-loans.mdx

```diff
@@ -0,0 +1,119 @@
+---
+sidebar_position: 5
+---
+
+# Flash loans
+
+The general setup for a flash loan requires:
+* A smart contract wallet (e.g. Safe Wallet).
+* A flash loan lender (e.g. Aave).
+* A use case for the funds (e.g. repaying a debt in Aave using a deposit as collateral).
+* The necessary `ERC-20` approvals in the wallet: to CoW's Vault Relayer contract, as well as any additional approvals related to the use case (e.g. Aave pools).
+* Some non-zero amount of the order's buy token in the wallet balance, to pass CoW Swap order validation.
+
+## Example: repay debt with collateral using flash loans on Sepolia
+
+We’ll walk through a concrete example of a debt repayment using flash loans on Sepolia.
+
+There is an [example order](https://explorer.cow.fi/sepolia/orders/0x86ff28a37bf70c549edfe753d899d44307b1b4c3ae943c6f5bdfe271942fd4f135ed9a9d1122a1544e031cc92fcc7ea599e28d9c67dc66ab) for a particular case on Sepolia, including the [corresponding onchain transaction](https://sepolia.etherscan.io/tx/0x6f4fc9b450e08dab0ab2d935e7c7dcade9e44e330e484657e5b1fadb6c6e8851).
+The example was executed using an [automated script](https://github.com/mrnaveira/cow-flashloan-tester) that places and signs a flash loan order on the Sepolia network.
+
+### Test setup
+The setup for the flash loan example is as follows:
+* A safe Wallet on address `0x35eD9A9D1122A1544e031Cc92fCC7eA599e28D9C`.
+* The Safe borrows 10,000 USDT on Aave using 400,000+ USDC as collateral.
+
+The order itself intends to:
+1. Request a flash loan to Aave of 5,000 USDT.
+2. Use those 5,000 to partially repay the 10,000 USDT debt on Aave.
+3. Withdraw 400,000 USDC from Aave.
```

Comment:
Why can we get back the majority of our collateral if we only repay half of our debt? Was that actually the setup of your debt position?